javascript-challenges.test.js

```diff
@@ -23,21 +35,77 @@ const secretCodeWord3 = "Eccentric"
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+const decode = (string) => {
+    // Create an empty string to store the decoded result
+    let decodedString = ''
+    // Loop through each character in the input string
+    for (let i = 0; i < string.length; i++) {
+        // Creat a const to single out each letter using the index
+        const letter = string[i]
+        // Use a conditional to have the desired number be retunred when the desired letter is indexed.  
+        if (letter === 'a') {
+            // Use the Addition assignment operator '+=' to have the const decodedString now equal the original string with the letters replaced.  
+          decodedString += '4'
+          //Repeat
+        } else if (letter === 'e') {
+          decodedString += '3'
+        } else if (letter === 'E') {
+            decodedString += '3'
+        }else if (letter === 'i') {
+          decodedString += '1'
+        } else if (letter === 'o') {
+          decodedString += '0'
+        } else {
+          decodedString += letter
+        }
+      }
+    
+      return decodedString
+    }
+
+// Pseudo code: ^^^
 
 // --------------------2) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
+describe ("poker", (string) => {
+    it("takes in an array of 5 numbers and determines whether or not the array is a full house.", () => {
+      expect(poker([5, 5, 5, 3, 3])).toEqual(true)
+      expect(poker([5, 5, 3, 3, 4])).toEqual(false)
+      expect(poker([5, 5, 5, 5, 4])).toEqual(false)
+      expect(poker([7, 2, 7, 2, 7])).toEqual(true)
+    }
+    )}
+  )
+//Output: Goodfail
+//ReferenceError: poker is not defined
+
+
+const arr1 = [5, 5, 5, 3, 3]
 // Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
+const arr2 = [5, 5, 3, 3, 4]
 // Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
+const arr3 = [5, 5, 5, 5, 4]
 // Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
+const arr4 = [7, 2, 7, 2, 7]
 // Expected output: true
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+const poker = (arr) => {
```

Comment:
Nice work, logic looks great!  If you are looking to stretch yourself, consider refactoring the first problem to use a higher order function.